fabric-entity-events-v1/src/main/java/net/fabricmc/fabric/api/entity/event/v1/ServerEntityWorldChangeEvents.java

```diff
@@ -75,6 +88,19 @@ public interface AfterEntityChange {
 		void afterChangeWorld(Entity originalEntity, Entity newEntity, ServerWorld origin, ServerWorld destination);
 	}
 
+	@FunctionalInterface
+	public interface AllowPlayerChange {
```

Comment:
Should this be limited to just players, other entities are allowed to use portals.

I think its also worth looking at 1.21.2 as I know they changed a lot of this teleportation code.

Reply:
Yes, this event is currently only used to restrict players when crossing dimensions; other entities are not affected. I believe there are very few scenarios where restricting entities from crossing dimensions would be necessary.

Reply:
Additionally, I will read through 1.21.2 as soon as possible, and I may be able to provide feedback by tomorrow.

Reply:
I have reviewed the relevant source code for 24w39a, and there are not many changes regarding the dimension-crossing code. Through testing, I can assure you that this event works correctly in the latest version as well.